Working version for discussions

triple-jay opened the pull request:
### Summary <!-- Required -->
Created the basic working version for the new discussions. 
<!-- Provide a general summary of your changes in the Title above -->
I worked on the new version of discussions that includes lots of new features, which are listed below.
<!-- Add your summary here -->
Here are the features that I worked on:
- Filter questions by if they are answered or unanswered
- Sort questions by upvote or most recent
- TAs/professors should be able to click the "Done" button to answer a question
- TAs/professors should be able to see the profile image and name of students who post questions, while normal students do not
- Answered questions should now have a resolved badge
- Updated the comment and upvote icons, and added new "sort by" arrows and resolved checkmark icons
- Normal students should not be able to comment on other student's questions
- No comments should be able to be added when discussion is over
<!-- Itemize bug fixes, new features, and other changes -->
<!-- Feel free to break this into sections, i.e. features, fixes, etc. -->

<!-- Add optional bullet points -->

### Test Plan <!-- Required -->
- Create an unanswered question and log in as a TA/professor, clicking "Done" to see if a question gets sent from unanswered questions to answered questions (can be checked by changing the filter slider)
- Create multiple questions, some answered and some unanswered, and filter by if they are answered or unanswered to see if it works properly
- Try sorting multiple questions by upvote or most recent and see if it works (most upvotes or most recent should be at the top, depending on what you sort by)
- Log in as a student to make sure you can't see the names and profile pictures of people asking questions
- Try logging in as a TA/professor to make sure you can see the names and profile pictures of people who ask questions
- Check that any question in answered questions has the resolved badge
- Try commenting as a student on other students' questions and make sure this does not work
- Find a previous discussion that is already over or wait until your discussion you are using to test is over, and make sure you can not add comments
- Make sure all the new and updated icons look good
<!-- Briefly describe how you test you changes. -->

### Notes <!-- Optional -->

<!--- List any important or subtle points, future considerations, or other items of note. -->

### Breaking Changes <!-- Optional -->

None

<!-- Uncomment any item below if it applies to your changes. -->

<!-- - Firebase schema change (requires migration plan)
<!-- - Firebase security policy change
<!-- - I updated existing types in `/src/components/types/`
<!-- - My changes requires a change to the documentation.
<!-- - Other change that could cause problems (Detailed in notes)

### Checklist

<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have updated the documentation accordingly.
- [ ] My PR adds a @ts-ignore

ngwattcos reviewed (changes requested):
Great job, Jeremy! I've tested your discussion features and all work. Nice job on implementing the toggles. I do have a couple nits and some bugs however:
Nits:
* could you speed up the toggle animation between Unanswered/Answered questions by 2x?
* the toggle between most recent/upvotes doesn't really depend on where you click. From a UX perspective, I would expect it to work like the larger toggle above, but it only works on the number of clicks. As such, if you click an odd number of times, you can coerce the component to flip to the state on the opposite side that you clicked on.


Some more serious issues:
* I think the voting functionality is bugged. There are 2 issues: 1) the order of posts when sorting by number of votes is inconsistent between different users, and 2) one of the posts can have up top 2 votes; the other post can have either 0 or 1 votes no matter which of the 2 users I vote with. I believe this may have to do with how the set of voters is stored.

ta's view:
![image](https://user-images.githubusercontent.com/18223523/114241131-b18d5980-9956-11eb-876e-c093cea7f912.png)

student's view:
![image](https://user-images.githubusercontent.com/18223523/114241582-60ca3080-9957-11eb-9d84-0a3725359528.png)

triple-jay commented:
Scott, as we discussed, I left off on fixing the voter discrepancy issue as we saw that it probably was not related to the data in Firebase and agreed that I should hold off on the issue. Other than that, I think my changes should fix the other addressed issues!
